src/testoptix/CMakeLists.txt

```diff
@@ -1,37 +1,99 @@
 # The 'textoptix' executable
 
+# Modified macro from liboslexec to compile the renderer support library
+# from CUDA to LLVM bitcode, and then embed the bitcode in a C++ file to
+# be compiled into the testoptix executable.
+MACRO ( LLVM_COMPILE llvm_src srclist )
+    GET_FILENAME_COMPONENT ( llvmsrc_we ${llvm_src} NAME_WE )
+    SET ( llvm_asm "${CMAKE_CURRENT_BINARY_DIR}/${llvmsrc_we}.s" )
+    SET ( llvm_bc "${CMAKE_CURRENT_BINARY_DIR}/${llvmsrc_we}.bc" )
+    SET ( llvm_bc_cpp "${CMAKE_CURRENT_BINARY_DIR}/${llvmsrc_we}.bc.cpp" )
+    if (VERBOSE)
+        MESSAGE (STATUS "LLVM_COMPILE (rend_lib) in=${llvm_src}")
+        MESSAGE (STATUS "LLVM_COMPILE (rend_lib) bc=${llvm_bc}")
+        MESSAGE (STATUS "LLVM_COMPILE (rend_lib) cpp=${llvm_bc_cpp}")
+    endif ()
+    SET ( ${srclist} ${${srclist}} ${llvm_bc_cpp} )
+    get_property (CURRENT_DEFINITIONS DIRECTORY PROPERTY COMPILE_DEFINITIONS)
+    if (VERBOSE)
+        message (STATUS "Current #defines are ${CURRENT_DEFINITIONS}")
+    endif ()
+    foreach (def ${CURRENT_DEFINITIONS})
+        set (LLVM_COMPILE_FLAGS ${LLVM_COMPILE_FLAGS} "-D${def}")
+    endforeach()
+    set (LLVM_COMPILE_FLAGS ${LLVM_COMPILE_FLAGS} ${SIMD_COMPILE_FLAGS} ${CSTD_FLAGS})
+
+    # Figure out what program we will use to make the bitcode.
+    if (NOT LLVM_BC_GENERATOR)
+        FIND_PROGRAM(LLVM_BC_GENERATOR NAMES "clang++" PATHS "${LLVM_DIRECTORY}/bin" NO_DEFAULT_PATH NO_CMAKE_SYSTEM_PATH NO_SYSTEM_ENVIRONMENT_PATH NO_CMAKE_ENVIRONMENT_PATH NO_CMAKE_PATH)
+    endif ()
+    # If that didn't work, look anywhere
+    if (NOT LLVM_BC_GENERATOR)
+        # Wasn't in their build, look anywhere
+        FIND_PROGRAM(LLVM_BC_GENERATOR NAMES clang++ llvm-g++)
+    endif ()
+
+    if (NOT LLVM_BC_GENERATOR)
+        message (FATAL_ERROR "You must have a valid llvm bitcode generator (clang++) somewhere.")
+    endif ()
+    if (VERBOSE)
+        message (STATUS "Using ${LLVM_BC_GENERATOR} to generate bitcode.")
+    endif()
+
+    # Command to turn the renderer library source file into LLVM bitcode .bc,
+    # then back into a C++ file with the bc embedded!
+    # This macro is set up to accept a .cu file, but could be adapted to accept
+    # a generic C++ file.
+    ADD_CUSTOM_COMMAND ( OUTPUT ${llvm_bc_cpp}
+      COMMAND ${LLVM_BC_GENERATOR}
+          "-I${OPTIX_INCLUDE_DIR}"
+          "-I${CUDA_INCLUDE_DIR}"
+          "-I${CMAKE_SOURCE_DIR}/cuda"
+          -D__CUDACC__ -D__CUDA_ARCH__=350
```

Comment:
I found that here, in the assembly of this command, I also needed:

    --cuda-path="${CUDA_TOOLKIT_ROOT_DIR}"